index.js

```diff
@@ -94,20 +148,48 @@ class FSMerge {
     return result;
   }
 
-  readDirSync(dirPath, options) {
+  readdirSync(dirPath, options) {
     let { _dirList } = this;
-    let result = [];
+    let result = [], errorCount = 0;
+    let fullDirPath = '';
     for (let i=0; i < _dirList.length; i++) {
       let { root } = getRootAndPrefix(_dirList[i]);
-      let fullDirPath = root + '/' + dirPath;
+      fullDirPath = root + '/' + dirPath;
       fullDirPath = fullDirPath.replace(/(\/|\/\/)$/, '');
       if(fs.existsSync(fullDirPath)) {
         result.push.apply(result, fs.readdirSync(fullDirPath, options));
+      } else {
+        errorCount += 1;
       }
     }
+    if (errorCount == _dirList.length) {
+      fs.readdirSync(fullDirPath);
+    }
     return [...new Set(result)];
   }
 
+  readdir(dirPath, callback) {
+    let result = [];
+    let promise = new Promise((res, rej) => {
+      try {
+        result = this.readdirSync(dirPath);
+      } catch(err) {
+        res({
+          err: err
+        });
+        return;
+      }
+      res({
+        err: null,
+        list: result
+      });
+      return;
+    });
+    promise.then((result) => {
```

Comment:
if this promise rejects, where does the error go?

Reply:
I'm not 100% sure using promises here makes this simply, any reason to not use the native  `fs.readdir` ?

Reply:
We can not use the native because we are accumulating paths from all the input paths which matches the directory path and return it.

Reply:
thats ok, but we need to mimic the native error handler semantics

Reply:
I could keep semantics almost same except throwing inside callback handling. Do you have any suggestion here?

Reply:
The simplest would be something like:

```js
let result, error;
try {
   result = this.readdirSync(dirPath);
 } catch (err) {
   error = err;
}

setTimeout(() => callback(error, result), 0);
```

Reply:
We want the readdir to be async this will make it sync. I have updated the code, which is works exactly like `fs.readdir`. Please have a look. I have added test to confirm that we are not deviating from `fs.readdir` too.